### DISCUSSION TITLE
What should I do with my UWP app?

### DISCUSSION THREAD

**Discussion by kinex** [microsoft/WindowsAppSDK#1218] [Q&A]
I am a developer of a relatively popular UWP app, and I would like to continue developing the app also in future. If I have understood correctly all this mess, UWP is dead and WindowsAppSDK is the way to go for Windows apps. So what does this mean for my UWP app? I guess I should port my app to WindowsAppSDK to give it a new future, right? Unfortunately I haven't had time to try WindowsAppSDK yet, so I have few related questions:

1. How big task is converting an UWP app to an app using WindowsAppSDK? Is it more like a rewrite or just porting?
2. My app needs access to namespace `Windows.ApplicationModel.Appointments` (APIs like `ShowEditNewAppointmentAsync` etc.). Is this namespace available if using WindowsAppSDK?
3. My app has also in-app purchases. Is this supported with WindowsAppSDK?
4. Are Live Tiles supported with WindowsAppSDK?
5. When I publish the upgraded app to Store, is it able to access the old UWP app data folder?

**Comment by BarcodeResource** [comment]
I have spend a few weeks to _test_ port our graphics UWP app to WinUI Desktop (reunion 0.8). I can get our app to launch with some code commented and the UI works. There are still still issues here and there. Our team feels like Dante's Neutral : neither here nor there, and I really hope communication could have been better for UWP developers like us (team morale is low).

A little summary from the notes I kept. 
1. For UI, it is mainly a change in namespace from Windows.UI.Xaml replace to Microsoft.UI.Xaml. To be fair, I think the SDK team did a great job here. I was expecting more issues but did not.
2. pickers need initializeWithWindow.Initialize(hwnd)
3. Small changes to make MainWindow to load existing Page. Need to use COM to set Window size and set icons.
4. Some changes to GridLength and Thickness. 
5. p = new Popup(); p.XamlRoot = this.Content.XamlRoot;
6. WebView to WebView2
7. printManager = PrintManager.GetForCurrentView();//.InteropServices.COMException 
8. await PrintManager.ShowPrintUIAsync();   //System.Runtime.InteropServices.COMException (0x80040155)
9. contacts = await page.PickContactsAsync();//System.InvalidOperationException
10. dispatcher.runasync
11. Code for listing Native Fonts changes 
12. Desktop Extensions changes
13. Windows.UI.ViewManagement.InputPane.GetForCurrentView().Showing throw element not found //Hide
14. Plan for MessageDialog to ContentDialog. Still supported.

Please note the above is a quick copy from my notes. some items may have been resolved along the way.
